gfop/get_food_counts.py

```diff
@@ -0,0 +1,134 @@
+    # Select food hierarchy levels.
+    sample_types = sample_types[f'sample_type_group{level}'] # remove this line to do file-level
+    # Match the GNPS job results to the food sample types.
+    sample_types_selected = sample_types.reindex(filenames)
+    sample_types_selected = sample_types_selected.dropna()
+    # Discard samples that occur less frequent than water (blank).
+    water_count = (sample_types_selected == 'water').sum()
+    sample_counts = sample_types_selected.value_counts()
+    sample_counts_valid = sample_counts.index[sample_counts > water_count]
+    sample_types_selected = sample_types_selected[
+        sample_types_selected.isin(sample_counts_valid)]
+    # Get sample counts at the specified level.
+    return sample_types_selected.value_counts()
+
+
+def get_dataset_food_counts(gnps_network, metadata, filename_col,
+                            sample_types, all_groups, some_groups,
+                            level, save_name):
+    """
+    Generate a table of food counts for a study dataset.
+
+    Args:
+        gnps_network (string): path to tsv file generated from classical molecular.
+                               networking with study dataset(s) and reference dataset.
+        metadata (string): path to sample metadata (comma- or tab-separated) file.
+                           Must contain a column with mzXML file names that match those used in the molecular networking job.
+        filename_col (string): name of column header containing file names.
+        sample_types (string): one of 'simple', 'complex', or 'None'.
+                               Simple foods are single ingredients while complex foods contain multiple ingredients.
+                               'None' will return both simple and complex foods.
+        all_groups (list): can contain 'G1', 'G2' to denote study spectrum files.
+        some_groups (list): can contain 'G3', 'G4' to denote reference spectrum files.
+        level (integer): indicates the level of the food ontology to use.
+                         One of 1, 2, 3, 4, 5, or 6.
+        save_name (string): path/file name to save food counts table.
+    Return:
+        A data frame. The data frame is also saved as a csv file to the specified directory.
+    Examples:
+        get_dataset_food_counts(gnps_network = 'METABOLOMICS-SNETS-V2-07f85565-view_all_clusters_withID_beta-main.tsv',
+                                metadata = 'sample_metadata.csv',
+                                filename_col = 'filenames',
+                                sample_types = 'simple',
+                                all_groups = ['G1'],
+                                some_groups = ['G4'],
+                                level = 5,
+                                save_name = "food_counts_L5")
+    """
+    food_counts, filenames = [], []
+    gnps_network = pd.read_csv(gnps_network, sep='\t')
+    sample_types = get_sample_types(sample_types)
+    if metadata.endswith('.csv'): delim = ','
```

Comment:
Weird if-else formatting; normally the statements should be on a new line. A one-liner would be: `delim = ',' if metadata.endswith('.csv') else '\t'

Reply:
Thanks for the tip!